src/picongpu/include/simulationControl/MySimulation.hpp

```diff
@@ -282,18 +281,25 @@ class MySimulation : public SimulationHelper<simDim>
 
         size_t freeGpuMem(0);
         Environment<>::get().EnvMemoryInfo().getMemoryInfo(&freeGpuMem);
-        freeGpuMem -= totalFreeGpuMemory;
+        if(freeGpuMem < reservedGpuMemorySize)
+        {
+            PMacc::log< picLog::MEMORY > ("%1% MiB free memory < %2% MiB required reserved memory")
+                % (freeGpuMem / 1024 / 1024) % (reservedGpuMemorySize / 1024 / 1024) ;
+            throw std::runtime_error("Cannot reserve enough memory");
```

Comment:
Isn't this what the log levels are for? If you have an obviously memory related error, you can turn on logging of memory related stuff to see what is going on.